subprojects/core/src/main/java/org/gradle/api/internal/project/ProjectStateRegistry.java

```diff
@@ -70,12 +71,12 @@ public interface ProjectStateRegistry {
     /**
      * Registers the projects of a build.
      */
-    void registerProjects(BuildState owner, ProjectRegistry<DefaultProjectDescriptor> projectRegistry);
+    void registerProjects(BuildState owner, SettingsInternal settingsInternal);
```

Comment:
❌  I'd avoid passing full settings here. The `BuildState` already exposes the `buildRootDir`, and we can expose the settings directory on it the same way. It's also a logic owner of that information.

Since we already have the `BuildState` as an argument, it should require less changes to callsites of these methods.